src/par_iter/internal.rs

```diff
@@ -7,14 +7,20 @@ use join;
 use super::IndexedParallelIterator;
 use super::len::*;
 
+pub trait ProducerContinuation<ITEM> {
```

Comment:
Is [this what you mean](https://en.wikipedia.org/wiki/Continuation) when you say 'continuation'? Might be handy to have a comment here.